internal/directives/argocd_updater.go

```diff
@@ -265,6 +265,12 @@ func (a *argocdUpdater) runPromotionStep(
 			continue
 		}
 
+		// Log the error, as it contains information about why we need to
+		// perform an update.
+		if err != nil {
+			logger.Debug(err.Error())
+		}
```

Comment:
I'm somewhat questioning whether an error explains why `mustUpdate` is `true`.

If successfully we've made the determination that we must update, wouldn't `err` be `nil`?

Reply:
I see. I hadn't quite realized we were treating mismatches as a sort of error and not _just_ returning true. All set.